core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -405,29 +440,54 @@ export class OptimizedSsrEngine {
         }
         this.log(
           `Rendering of ${request?.originalUrl} was not able to complete. This might cause memory leaks!`,
+          { request },
           false
         );
       }, this.ssrOptions?.maxRenderTime ?? 300000); // 300000ms == 5 minutes
 
-    this.log(`Rendering started (${request?.originalUrl})`);
+    this.log(`Rendering started (${request?.originalUrl})`, { request });
     this.renderingCache.setAsRendering(renderingKey);
     this.currentConcurrency++;
 
+    options = {
+      ...options,
+      providers: [
+        {
+          provide: serverLoggerToken,
+          useValue: this.logger,
+        },
+        {
+          provide: loggerEnabled,
+          useValue: !!this.ssrOptions?.logger,
+        },
+      ],
+    };
+
     this.expressEngine(filePath, options, (err, html) => {
       if (!maxRenderTimeout) {
         // ignore this render's result because it exceeded maxRenderTimeout
         this.log(
           `Rendering of ${request.originalUrl} completed after the specified maxRenderTime, therefore it was ignored.`,
+          { request },
           false
         );
         return;
       }
       clearTimeout(maxRenderTimeout);
 
-      this.log(`Rendering completed (${request?.originalUrl})`);
+      this.log(`Rendering completed (${request?.originalUrl})`, {
+        request,
+      });
       this.currentConcurrency--;
 
       renderCallback(err, html);
     });
   }
+
+  private initLogger(ssrOptions: SsrOptimizationOptions | undefined) {
+    if (typeof ssrOptions?.logger === 'boolean') {
+      return new ExpressServerLogger();
+    }
+    return ssrOptions?.logger ?? new ServerLogger();
```

Comment:
[MEDIUM]

I'm aware that the interface doesn't allow for `false`, but in the future we might add it (since next major). So let's handle `false` value simply here, to be future proof - when it's false, use the base ServerLogger.

What about:
```suggestion
    if (ssrOptions?.logger === true) {
      return new ExpressServerLogger();
    }
    return ssrOptions?.logger || new ServerLogger();
```
let's also add an unit test for false